src/binary/decimal.rs

```diff
@@ -34,12 +46,21 @@ where
     W: Write,
 {
     fn encode_decimal(&mut self, decimal: &Decimal) -> IonResult<()> {
+        if decimal == &DECIMAL_POSITIVE_ZERO {
+            return Ok(());
+        }
```

Comment:
My intention was to implement this from the spec:

> If the value is 0. (aka 0d0) then L is zero, there are no length or representation fields, and the entire value is encoded as the single byte 0x50.

I took that to be structural, i.e. 0dE only for E = 0. There aren't test cases for this, which seems to be a miss.

In terms of implementation.. 

```rust
impl PartialEq for Decimal {
    fn eq(&self, other: &Self) -> bool {
        self.cmp(other) == Ordering::Equal
    }
}
```

and, the relevant section in `cmp` is:

```rust
        if d1.exponent < d2.exponent {
            Self::compare_scaled_magnitudes(d1, d2)
        } else {
            Self::compare_scaled_magnitudes(d2, d1)
        }
```

I believe that will disambiguate 0d100 from 0d0 from 0d10.

Reply:
Pushed a commit with a test to show how I understood it.

Reply:
Yes, that verifies what I was expecting, we have a structural comparison here so that is what we want for what you're doing.  What is interesting (and out of scope for your PR) is that most people will not expect this behavior as they will expect value comparison.  I think this is basically related to #220, we should really push this structural stuff into its own trait.